src/video/n3ds/SDL_n3dsvideo.c

```diff
@@ -0,0 +1,163 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2022 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "../../SDL_internal.h"
+
+#ifdef SDL_VIDEO_DRIVER_N3DS
+
+#include "../SDL_sysvideo.h"
+#include "SDL_n3dsevents_c.h"
+#include "SDL_n3dsframebuffer_c.h"
+#include "SDL_n3dsswkb.h"
+#include "SDL_n3dsvideo.h"
+
+#define N3DSVID_DRIVER_NAME "n3ds"
+
+SDL_FORCE_INLINE void AddN3DSDisplay(gfxScreen_t screen);
+
+static int N3DS_VideoInit(_THIS);
+static void N3DS_VideoQuit(_THIS);
+static void N3DS_GetDisplayModes(_THIS, SDL_VideoDisplay *display);
+static int N3DS_CreateWindow(_THIS, SDL_Window *window);
+static void N3DS_DestroyWindow(_THIS, SDL_Window *window);
+
+/* N3DS driver bootstrap functions */
+
+static void
+N3DS_DeleteDevice(SDL_VideoDevice *device)
+{
+    SDL_free(device->displays);
+    SDL_free(device->driverdata);
+    SDL_free(device);
+}
+
+static SDL_VideoDevice *
+N3DS_CreateDevice(void)
+{
+    SDL_VideoDevice *device = (SDL_VideoDevice *) SDL_calloc(1, sizeof(SDL_VideoDevice));
+    if (!device) {
+        SDL_OutOfMemory();
+        return (0);
+    }
+
+    device->VideoInit = N3DS_VideoInit;
+    device->VideoQuit = N3DS_VideoQuit;
+
+    device->GetDisplayModes = N3DS_GetDisplayModes;
```

Comment:
You should also implement `GetDisplayBounds()` to indicate the correct positioning of the displays, since the video subsystem will assume the displays are arranged left to right by default.